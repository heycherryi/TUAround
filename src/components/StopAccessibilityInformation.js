import React from 'react';
import PropTypes from 'prop-types';
import { faWheelchair, faQuestionCircle, faBan } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export function StopAccessibilityInformation({ stop }) {
  let wheelchair_boarding = '';
  switch (stop.wheelchair_boarding) {
    case '1':
      wheelchair_boarding = (
        <span className="me-2">
          <FontAwesomeIcon icon={faWheelchair} fixedWidth={true}></FontAwesomeIcon> Some vehicles at this stop can be boarded by a rider in a wheelchair.
        </span>
      );
      break;

    case '2':
      wheelchair_boarding = (
        <>
          <span className="fa-layers fa-fw me-1">
            <FontAwesomeIcon icon={faWheelchair} fixedWidth={true} transform="shrink-7"></FontAwesomeIcon>
            <FontAwesomeIcon icon={faBan} fixedWidth={true} className="text-danger"></FontAwesomeIcon>
          </span>
          Wheelchair boarding is not possible at this stop.
        </>
      );
      break;

    case '0':
    default:
      wheelchair_boarding = (
        <>
          <span className="fa-layers fa-fw me-1">
            <FontAwesomeIcon icon={faQuestionCircle} fixedWidth={true} className="text-warning"></FontAwesomeIcon>
          </span>
          No accessibility information for the stop.
        </>
      );
      break;
  }

  return wheelchair_boarding;
}

StopAccessibilityInformation.propTypes = {
  stop: PropTypes.object.isRequired,
};

export default StopAccessibilityInformation;
