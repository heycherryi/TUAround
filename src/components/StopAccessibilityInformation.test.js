/* globals test, expect */

import React from 'react';
import { render } from '@testing-library/react';
import StopAccessibilityInformation from './StopAccessibilityInformation';

const stopFixture1 = require('../fixtures/stop-MCC5_10.json');
const stopFixture2 = require('../fixtures/stop-DOVDICWF.json');
const stopFixture3 = require('../fixtures/stop-10AHERNN.json');

test('renders StopAccessibilityInformation with wheelchair boarding (1)', () => {
  const { container } = render(
    <StopAccessibilityInformation stop={stopFixture1} />,
  );
  expect(container).toHaveTextContent('Some vehicles at this stop can be boarded by a rider in a wheelchair.');
  expect(container).toMatchSnapshot();
});

test('renders StopAccessibilityInformation declared unknown wheelchair boarding (0)', () => {
  const { container } = render(
    <StopAccessibilityInformation stop={stopFixture2} />,
  );
  expect(container).toMatchSnapshot();
  expect(container).toHaveTextContent('No accessibility information for the stop.');
});

test('renders StopAccessibilityInformation with wheelchair boarding unset (null)', () => {
  const { container } = render(
    <StopAccessibilityInformation stop={stopFixture3} />,
  );
  expect(container).toHaveTextContent('No accessibility information for the stop.');
  expect(container).toMatchSnapshot();
});

test('renders StopAccessibilityInformation with wheelchair boarding unavailable (2)', () => {
  const stopFixture4 = stopFixture1;
  stopFixture4.wheelchair_boarding = '2';
  const { container } = render(
    <StopAccessibilityInformation stop={stopFixture4} />,
  );
  expect(container).toHaveTextContent('Wheelchair boarding is not possible at this stop.');
  expect(container).toMatchSnapshot();
});
