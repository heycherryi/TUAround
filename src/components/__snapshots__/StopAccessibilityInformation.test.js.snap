// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StopAccessibilityInformation declared unknown wheelchair boarding (0) 1`] = `
<div>
  <span
    class="fa-layers fa-fw me-1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-question fa-fw text-warning"
      data-icon="circle-question"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24V250.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1H222.6c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
        fill="currentColor"
      />
    </svg>
  </span>
  No accessibility information for the stop.
</div>
`;

exports[`renders StopAccessibilityInformation with wheelchair boarding (1) 1`] = `
<div>
  <span
    class="me-2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-wheelchair fa-fw "
      data-icon="wheelchair"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M192 96a48 48 0 1 0 0-96 48 48 0 1 0 0 96zM120.5 247.2c12.4-4.7 18.7-18.5 14-30.9s-18.5-18.7-30.9-14C43.1 225.1 0 283.5 0 352c0 88.4 71.6 160 160 160c61.2 0 114.3-34.3 141.2-84.7c6.2-11.7 1.8-26.2-9.9-32.5s-26.2-1.8-32.5 9.9C240 440 202.8 464 160 464C98.1 464 48 413.9 48 352c0-47.9 30.1-88.8 72.5-104.8zM259.8 176l-1.9-9.7c-4.5-22.3-24-38.3-46.8-38.3c-30.1 0-52.7 27.5-46.8 57l23.1 115.5c6 29.9 32.2 51.4 62.8 51.4h5.1c.4 0 .8 0 1.3 0h94.1c6.7 0 12.6 4.1 15 10.4L402 459.2c6 16.1 23.8 24.6 40.1 19.1l48-16c16.8-5.6 25.8-23.7 20.2-40.5s-23.7-25.8-40.5-20.2l-18.7 6.2-25.5-68c-11.7-31.2-41.6-51.9-74.9-51.9H282.2l-9.6-48H336c17.7 0 32-14.3 32-32s-14.3-32-32-32H259.8z"
        fill="currentColor"
      />
    </svg>
     Some vehicles at this stop can be boarded by a rider in a wheelchair.
  </span>
</div>
`;

exports[`renders StopAccessibilityInformation with wheelchair boarding unavailable (2) 1`] = `
<div>
  <span
    class="fa-layers fa-fw me-1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-wheelchair fa-fw "
      data-icon="wheelchair"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="transform-origin: 0.5em 0.5em;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        transform="translate(256 256)"
      >
        <g
          transform="translate(0, 0)  scale(0.5625, 0.5625)  rotate(0 0 0)"
        >
          <path
            d="M192 96a48 48 0 1 0 0-96 48 48 0 1 0 0 96zM120.5 247.2c12.4-4.7 18.7-18.5 14-30.9s-18.5-18.7-30.9-14C43.1 225.1 0 283.5 0 352c0 88.4 71.6 160 160 160c61.2 0 114.3-34.3 141.2-84.7c6.2-11.7 1.8-26.2-9.9-32.5s-26.2-1.8-32.5 9.9C240 440 202.8 464 160 464C98.1 464 48 413.9 48 352c0-47.9 30.1-88.8 72.5-104.8zM259.8 176l-1.9-9.7c-4.5-22.3-24-38.3-46.8-38.3c-30.1 0-52.7 27.5-46.8 57l23.1 115.5c6 29.9 32.2 51.4 62.8 51.4h5.1c.4 0 .8 0 1.3 0h94.1c6.7 0 12.6 4.1 15 10.4L402 459.2c6 16.1 23.8 24.6 40.1 19.1l48-16c16.8-5.6 25.8-23.7 20.2-40.5s-23.7-25.8-40.5-20.2l-18.7 6.2-25.5-68c-11.7-31.2-41.6-51.9-74.9-51.9H282.2l-9.6-48H336c17.7 0 32-14.3 32-32s-14.3-32-32-32H259.8z"
            fill="currentColor"
            transform="translate(-256 -256)"
          />
        </g>
      </g>
    </svg>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-ban fa-fw text-danger"
      data-icon="ban"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M367.2 412.5L99.5 144.8C77.1 176.1 64 214.5 64 256c0 106 86 192 192 192c41.5 0 79.9-13.1 111.2-35.5zm45.3-45.3C434.9 335.9 448 297.5 448 256c0-106-86-192-192-192c-41.5 0-79.9 13.1-111.2 35.5L412.5 367.2zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256z"
        fill="currentColor"
      />
    </svg>
  </span>
  Wheelchair boarding is not possible at this stop.
</div>
`;

exports[`renders StopAccessibilityInformation with wheelchair boarding unset (null) 1`] = `
<div>
  <span
    class="fa-layers fa-fw me-1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-circle-question fa-fw text-warning"
      data-icon="circle-question"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24V250.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1H222.6c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
        fill="currentColor"
      />
    </svg>
  </span>
  No accessibility information for the stop.
</div>
`;
